src/main/scala/com/fulcrumgenomics/fasta/SequenceDictionary.scala

```diff
@@ -0,0 +1,206 @@
+        val (refName: String, start: Int, end: Int) = FgBioDef.parseRange(range)
+        val locus: AlternateLocus = {
+          if (refName == "=") AlternateLocus(refName=this.name, start=start.toInt, end=end.toInt)
+          else AlternateLocus(refName=refName, start=start.toInt, end=end.toInt)
+        }
+        Some(locus)
+    }
+  }
+  @inline final def md5: Option[String] = this.get(Keys.Md5)
+  @inline final def md5Int: Option[BigInt] = this.md5.map(BigInt(_, 16))
+  @inline final def assembly: Option[String] = this.get(Keys.Assembly)
+  @inline final def uri: Option[String] = this.get(Keys.Uri)
+  @inline final def species: Option[String] = this.get(Keys.Species)
+  @inline final def description: Option[String] = this.get(Keys.Description)
+  @inline final def topology: Option[Topology] = {
+    this.get(Keys.Topology).flatMap(tp => Topology.values.find(_.name == tp))
+  }
+
+  /** Returns true if the the sequences share a common reference name (including aliases), have the same length, and
+    * the same MD5 if both have MD5s. */
+  def sameAs(that: SequenceMetadata): Boolean = {
+    val md5Match = (this.md5Int, that.md5Int) match {
+      case (Some(thisMd5Int), Some(thatMd5Int)) => thisMd5Int == thatMd5Int
+      case _                                    => true
+    }
+
+    this.length == that.length &&
+      (this.name == that.name || this.allNames.exists(that.allNames.contains)) &&
+      md5Match
+  }
+}
+
+object SequenceDictionary {
+  /** Builds a sequence dictionary from one or more sequence metadatas.  This will set the sequence index. */
+  def apply(info: SequenceMetadata*): SequenceDictionary = {
+    val infos = info.zipWithIndex.map { case (info, index) =>
+      if (info.index != index) info.copy(index=index) else info
+    }.toIndexedSeq
+    SequenceDictionary(infos=infos)
+  }
+}
+
+/** Contains an ordered collection of sequences. */
+case class SequenceDictionary(infos: IndexedSeq[SequenceMetadata]) extends Iterable[SequenceMetadata] {
+
+  private val mapping: Map[String, SequenceMetadata] = {
+    // validates that the same name is not present twice across allNames (so includes aliases)
+    val names: mutable.Set[String] = mutable.HashSet[String]()
+    infos.iterator.zipWithIndex.flatMap { case (info, index) =>
+      require(info.index == index, s"Infos must be given with index set correctly. See ${index}th with name: $names")
+      info.allNames.map { name =>
+        require(!names.contains(name), f"Found duplicate sequence name: $name")
+        names.add(name)
+        name -> info
+      }
+    }.toMap
+  }
+
+  def apply(name: String): SequenceMetadata = this.mapping(name)
+  def get(name: String): Option[SequenceMetadata] = this.mapping.get(name)
+  def contains(name: String): Boolean = this.mapping.contains(name)
+  def apply(index: Int): SequenceMetadata = this.infos(index)
+  override def iterator: Iterator[SequenceMetadata] = this.infos.iterator
+  def length: Int = this.infos.length
```

Comment:
A thought for our future selves.... this would be a nice place to add methods like `genomeSize` that gives you the total of all the sequences.